src/services/blocks/BlocksService.ts

```diff
@@ -294,6 +328,80 @@ export class BlocksService extends AbstractService {
 		return response;
 	}
 
+	/**
+	 * Fetch `payment_queryFeeDetails`.
+	 *
+	 * @param extHex
+	 * @param previousBlockHash
+	 * @param extrinsicSuccessWeight
+	 * @param estWeight
+	 */
+	private async fetchQueryFeeDetails(
+		extHex: `0x${string}`,
+		previousBlockHash: BlockHash,
+		extrinsicSuccessWeight: Weight,
+		estWeight: Weight
+	): Promise<string> {
+		const { api } = this;
+		const { inclusionFee } = await api.rpc.payment.queryFeeDetails(
+			extHex,
+			previousBlockHash
+		);
+		const finalPartialFee = this.calcPartialFee(
+			extrinsicSuccessWeight,
+			estWeight,
+			inclusionFee
+		);
+
+		return finalPartialFee;
+	}
+
+	/**
+	 * Retrieve the blockHash for the previous block to the one getting queried.
+	 * If the block is the geneisis hash it will return the same blockHash.
+	 *
+	 * @param blockNumber The blockId being queried
+	 */
+	private async fetchPreviousBlockHash(
+		blockNumber: Compact<BlockNumber>
+	): Promise<BlockHash> {
+		const { api } = this;
+
+		const num = blockNumber.toBn();
+		return num.isZero()
+			? await api.rpc.chain.getBlockHash(num)
+			: await api.rpc.chain.getBlockHash(num.sub(new BN(1)));
+	}
+
+	/**
+	 * Calculate the partialFee for an extrinsic. This uses `calc_partial_fee` from `@substrate/calc`.
+	 * Please reference
```

Comment:
Is this comment unfinished?

Reply:
complete the comment. I just needed to reference the user to check out the rust code to see docs on the algorithm we use.